examples/typescript/informer/informer.ts

```diff
@@ -0,0 +1,26 @@
+import * as k8s from '@kubernetes/client-node';
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+
+const listFn = (fn) => {
+    k8sApi.listNamespacedPod('default')
+        .then((res) => {
+            const podList = res.body;
+            fn(podList.items, podList.metadata!.resourceVersion);
```

Comment:
Why does this example use `!.resourceVersion` and the one above uses `podList.metadata.resourceVersion`

Reply:
This is the first time I am reading about informers, and this is epic! So excuse my possibly silly questions.

Is there a reason for using callback style here vs just promise style where we return the values? It makes it a little hard to read and I think it complicates the public API a little.

I was reading https://www.firehydrant.io/blog/stay-informed-with-kubernetes-informers/ and the go client's interface seems way more natural because it removes a bunch of this boilerplate logic.

and I think it makes sense to abstract as MUCH as possible about these functions (if we can).

Reply:
I will refactor to promises.

What additional boilerplate do you think we can remove? A lot of the client-go code is code-generated to cover all API objects....

Reply:
It was mostly related to the functions/callbacks and having to do error handling. If we just return the promise that is handled automatically for us. (I believe).

I get we can't get the same level of API as the go client because we aren't there so we will have to have the internal function to get the updated data, just wanted to try to make it cleaner even if its slightly, and I believe promises could get us 90% there.

It isn't 100% clear to me what value's have to be passed into the function is it just the data and version?

Reply:
I kinda see it working like this 

```
return k8sApi.listNamespacedPod('default')
        .then((res) => Promise.resolve.(res.body.items, res.body.metadata!.resourceVersion))
```

I guess that be generic enough that the users could just switch out the `k8sApi.listFunction` and the first param.

sudo code might not actually work.

Reply:
I refactored this to be promises, please take a look. It looks a lot cleaner, imho, so thanks for the suggestion. I had to augment the tests a little too....